### DISCUSSION TITLE
I have attempted to install AlphaCodium and configured the LLM's model and key according to the instructions. Does this mean I can now connect to the LLM? Is there no need to configure a base_url?

### DISCUSSION THREAD

**Discussion by zte-yangjizhe** [Codium-ai/AlphaCodium#12] [Q&A] [answered]
I have attempted to install AlphaCodium and configured the LLM's model and key according to the instructions. Does this mean I can now connect to the LLM? Is there no need to configure a base_url?

**Comment by mrT23** [comment]
what do you mean by "base_rul" ??
you are working with openai models. Their interface is defined here:
https://github.com/Codium-ai/AlphaCodium/blob/main/alpha_codium/llm/ai_handler.py#L108

**Reply by zte-yangjizhe** [reply]
Oh, sorry, what I meant was api_base.
I have two questions: First, does AlphaCodium support all of OpenAI's LLMs? I used an internally developed model called CodeLlama's embedded model, but it seems to fail when running. 
![image](https://github.com/Codium-ai/AlphaCodium/assets/152255029/3b5bd724-64ab-4d37-8695-4648ada48edb)

Second, what is the minimum Python version requirement? I noticed that the Dockerfile specifies version 3.9.

**Reply by zte-yangjizhe** [reply]
I tried installing Python 3.9.1, but encountered the following error during execution
![image](https://github.com/Codium-ai/AlphaCodium/assets/152255029/54e31f7d-6371-41ba-ac9d-16b3296aae29)

**Reply by zte-yangjizhe** [reply]
and more：
![image](https://github.com/Codium-ai/AlphaCodium/assets/152255029/8079b4e5-6cdc-42b9-aa38-eb1a055d07aa)

**Reply by zte-yangjizhe** [reply]
![image](https://github.com/Codium-ai/AlphaCodium/assets/152255029/5f81384e-a021-4aed-b62e-fa7111647221)

**Comment by mrT23** [comment]
"AlphaCodium" (i.e litellm) supports any openai model from:
https://platform.openai.com/docs/models/gpt-4-and-gpt-4-turbo
https://platform.openai.com/docs/models/gpt-3-5

Regarding other models, you need to **debug** and edit if needed whats happening in:
https://github.com/Codium-ai/AlphaCodium/blob/main/alpha_codium/llm/ai_handler.py#L108

**Reply by zte-yangjizhe** [reply]
Can you provide me with a list of the successful version information you have tested? For example, versions of OpenAI, litellm, OpenSSL, etc.

**Reply by mrT23** [reply]
No.
And thats not the problem you experience.
Try first with a standard openai model, and make sure it works. it would. 
than debug your problem with your custom model

all the best

**Reply by zte-yangjizhe** [reply, marked as answer]
Thank you, I have successfully executed it
